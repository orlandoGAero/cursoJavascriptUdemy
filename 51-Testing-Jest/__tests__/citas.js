import Citas from '../js/classes/Citas';

describe('Testing a citas', () => {

    const citas = new Citas();

    const id = Date.now();

    test('Agregar cita', () => {
        const citaObj = {
            id,
            mascota: 'Hachi',
            propietario: 'Orlando',
            telefono: '7224514568',
            fecha: '10-03-2021',
            hora:'18:00',
            sintomas: 'no bebe'
        }

        citas.agregarCita(citaObj);

        // prueba
        expect(citas).toMatchSnapshot();
    });

    test('Editar Cita', () => {
        const citaObj = {
            id,
            mascota: 'Rabito',
            propietario: 'Irving',
            telefono: '7224514568',
            fecha: '10-03-2021',
            hora:'18:00',
            sintomas: 'no corre'
        }

        citas.editarCita(citaObj);

        expect(citas).toMatchSnapshot();
    });

    test('Eliminar cita', () => {
        citas.eliminarCita(id);

        expect(citas).toMatchSnapshot();
    })
})