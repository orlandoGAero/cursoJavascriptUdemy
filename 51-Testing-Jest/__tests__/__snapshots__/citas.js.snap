// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing a citas Agregar cita 1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "10-03-2021",
      "hora": "18:00",
      "id": 1614986232955,
      "mascota": "Hachi",
      "propietario": "Orlando",
      "sintomas": "no bebe",
      "telefono": "7224514568",
    },
  ],
}
`;

exports[`Testing a citas Editar Cita 1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "10-03-2021",
      "hora": "18:00",
      "id": 1614986232955,
      "mascota": "Rabito",
      "propietario": "Irving",
      "sintomas": "no corre",
      "telefono": "7224514568",
    },
  ],
}
`;

exports[`Testing a citas Eliminar cita 1`] = `
Citas {
  "citas": Array [],
}
`;
